### DISCUSSION TITLE
How to search with many files of queries

### DISCUSSION THREAD

**Discussion by Wenze7** [castorini/pyserini#1374] [Q&A] [answered]
I have too many queries to search, and I want to use multi threads, how should i do?

**Comment by lintool** [comment, marked as answer]
Use:

```
python -m pyserini.search.lucene \
  --threads xxx --batch-size yyy
...
```

See https://castorini.github.io/pyserini/2cr/msmarco-v1-passage.html for sample invocations.

**Reply by Wenze7** [reply]
Thanks!

**Comment by Hannibal046** [comment]
Hi, what would you suggest to set the number of thread and batch size?

**Reply by lintool** [reply]
Suggestion: set the number to threads to the number of cores you want to use; set the batch size to something like 8 * number of threads.
